include/stratosphere/kvdb/kvdb_memory_key_value_store.hpp

```diff
@@ -0,0 +1,566 @@
+/*
+ * Copyright (c) 2018-2019 Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <algorithm>
+#include <switch.h>
+#include <sys/stat.h>
+#include "kvdb_auto_buffer.hpp"
+#include "kvdb_archive.hpp"
+#include "kvdb_bounded_string.hpp"
+#include "kvdb_memory_key_value_store.hpp"
+
+namespace sts::kvdb {
+
+    template<class Key>
+    class MemoryKeyValueStore {
+        static_assert(std::is_pod<Key>::value, "KeyValueStore Keys must be pod!");
+        NON_COPYABLE(MemoryKeyValueStore);
+        NON_MOVEABLE(MemoryKeyValueStore);
+        public:
+            /* Subtypes. */
+            class Entry {
+                private:
+                    Key key;
+                    void *value;
+                    size_t value_size;
+                public:
+                    constexpr Entry(const Key &k, void *v, size_t s) : key(k), value(v), value_size(s) { /* ... */ }
+
+                    const Key &GetKey() const {
+                        return this->key;
+                    }
+
+                    template<typename Value = void>
+                    Value *GetValuePointer() {
+                        /* Size check. Note: Nintendo does not size check. */
+                        if constexpr (!std::is_same<Value, void>::value) {
```

Comment:
`std::type_trait<args...>::value` can usually be written as `std::type_trait_v<args...>`.

Reply:
yeah, style preference.